### PR TITLE
feat(Core/Config): CONFIG_INTERVAL_SAVE

### PR DESCRIPTION
**Pull request by Kitzunu**
## CHANGES PROPOSED:
-  implement(Core/Config): CONFIG_INTERVAL_SAVE

## TESTS PERFORMED:
- Builds

## HOW TO TEST THE CHANGES:

Honestly I don't know how it saves, so if you figure that out, then you can see if the timer works correctly :D

## Target branch(es):
- [x] Master

 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by BarbzYHOOL** [comment]
ported from TC?

**Comment by Kitzunu** [comment]
Basically what TC uses but adapted to AC. I haven't tested it myself so dont want to PR it

**Comment by pklloveyou** [comment]
Tested.

**Comment by pklloveyou** [comment]
My server crashed twice. I don't know if this is the reason. Hope someone tests it

**Comment by FrancescoBorzi** [comment]
conflicts and possible crash

**Comment by pklloveyou** [comment]
I'll test it on my server for a week,

**Comment by pklloveyou** [comment]
It can be merged and tested for a week without problems

File: src/server/scripts/Commands/cs_misc.cpp [hunk 1]
```diff
@@ -1086,9 +1086,26 @@ class misc_commandscript : public CommandScript
...
+            else
+                player->SaveToDB(true, false);
+            handler->SendSysMessage(LANG_PLAYER_SAVED);
+            return true;
+        }
+
+        // save if the player has last been saved over 20 seconds ago
+        uint32 saveInterval = sWorld->getIntConfig(CONFIG_INTERVAL_SAVE);
+        if (saveInterval == 0 || (saveInterval > 20 * IN_MILLISECONDS && player->GetSaveTimer() <= saveInterval - 20 * IN_MILLISECONDS))
+            player->SaveToDB(true, false);
```
Comment (FrancescoBorzi) [inline src/server/scripts/Commands/cs_misc.cpp:1107]: can you please add brackets to all these IFs ?

Comment (Kitzunu) [reply src/server/scripts/Commands/cs_misc.cpp:1107]: Ye when I got time

**Pull request merged**